tests/pipeline/test_resources_evaluation.py

```diff
@@ -212,99 +215,200 @@ async def async_resource1():
+
+    assert len(result) == sum(item_counts)
+
+    # Check extracted results from each resource
+    chunked_results = []
+    start_range = 0
+    for item_range in item_ranges:
+        chunked_results.append([item for item in result if item in item_range])
+
+    for i, chunk in enumerate(chunked_results):
+        # All items are included
+        assert len(chunk) == item_counts[i]
+        assert len(set(chunk)) == len(chunk)
+        # Items are extracted in order per resource
+        assert chunk == sorted(chunk)
+
+
+def test_test_parallelized_transformers() -> None:
+    item_count = 6
+
+    @dlt.resource(parallelized=True)
+    def pos_data():
+        for i in range(1, item_count + 1):
+            time.sleep(0.1)
+            yield i
+
+    @dlt.resource(parallelized=True)
+    def neg_data():
+        time.sleep(0.05)
+        for i in range(-1, -item_count - 1, -1):
+            time.sleep(0.1)
+            yield i
+
+    @dlt.transformer(parallelized=True)
+    def multiply(item):
+        time.sleep(0.05)
+        exec_order.append("+" if item > 0 else "-")
+        yield item * 10
+
+    @dlt.source
+    def some_source():
+        return [
+            neg_data | multiply.with_name("t_a"),
+            pos_data | multiply.with_name("t_b"),
+        ]
+
+    exec_order: List[str] = []
+    result = list(some_source())
+
+    expected_result = [i * 10 for i in range(-item_count, item_count + 1)]
+    expected_result.remove(0)
+
+    assert sorted(result) == expected_result
+    assert exec_order == ["+", "-"] * item_count
+
+    @dlt.transformer(parallelized=True)  # type: ignore[no-redef]
+    def multiply(item):
+        # Transformer that is not a generator
+        time.sleep(0.05)
+        exec_order.append("+" if item > 0 else "-")
+        return item * 10
+
+    @dlt.source  # type: ignore[no-redef]
+    def some_source():
+        return [
+            neg_data | multiply.with_name("t_a"),
+            pos_data | multiply.with_name("t_b"),
+        ]
+
+    exec_order = []
+
+    result = list(some_source())
+
+    expected_result = [i * 10 for i in range(-item_count, item_count + 1)]
+    expected_result.remove(0)
+
+    assert sorted(result) == expected_result
+    assert exec_order == ["+", "-"] * item_count
+
+
+def test_parallelized_bare_generator() -> None:
+    def pos_data():
+        for i in range(1, 6):
+            time.sleep(0.1)
+            yield i
+
+    def neg_data():
+        time.sleep(0.05)
+        for i in range(-1, -6, -1):
+            time.sleep(0.1)
+            yield i
+
+    @dlt.source
+    def some_source():
+        return [
+            # Resources created from generators directly (not generator functions) can be parallelized
+            dlt.resource(pos_data(), parallelized=True, name="pos_data"),
+            dlt.resource(neg_data(), parallelized=True, name="neg_data"),
+        ]
+
+    result = list(some_source())
+
+    assert set(result) == {1, 2, 3, 4, 5, -1, -2, -3, -4, -5}
```

Comment:
I'm not checking the order here since it's not that guaranteed, just that all items are here. But I updated all the tests to check thread idents ( >1 thread and not main thread)  
Would be nice to check execution order is roughly interleaved but tricky without making flaky tests